submission/distributor.go

```diff
@@ -68,12 +74,68 @@ func (d *Distributor) Run(ctx context.Context) {
 	}
 }
 
-// refreshRoots requests roots from Logs and updates local copy if necessary.
+// refreshRoots requests roots from Logs and updates local copy.
 func (d *Distributor) refreshRoots(ctx context.Context) {
-	// TODO(Mercurrent) add implementation.
+	type RootsResult struct {
+		LogURL string
+		Roots  []*x509.Certificate
+		Err    error
+	}
+	ch := make(chan RootsResult, len(d.logClients))
+
+	rctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
nit: May be useful to make this timeout configurable, e.g. by flag.

Reply:
Move into a constant for now.